include/capmap.h

```diff
@@ -108,7 +108,12 @@ class Mapper {
     update_self_ranges();
     if (cheri_tag_get(cap)) {
       roots_.push_back(std::make_pair(name, cap));
-      scan(cap);
+      try {
+        scan(cap);
+      } catch (int n) {
+        fprintf(stderr, " from root %s at depth %d\n", name, n);
+        abort();
+      }
```

Comment:
The reason I use exceptions is to unwind the stack and print each capability which led to the poisoned access. Without some form of unwinding, this would be difficult.

Reply:
I have an idea to carry a kind of breadcrumb trail of capabilities around whilst recursing, because that would allow a deeper (if much more verbose) analysis. Right now, this tool can tell you that some memory is reachable. However, I'd like it to be able to trace the (capability) path that gets it there, since you need that to fix the problem.

I think when (and if) that's done, these exceptions won't be required. In the meantime, let's keep them, since they do serve a useful purpose.

Reply:
It sounds to me like that might be a good follow-up PR? We don't have to solve every problem in one PR :)

Reply:
No, let's save that for another PR.